### ISSUE TITLE
Workshop ID with New Mod

### ISSUE THREAD

**Issue by yenyang** [PropaneDragon/CimTools#3]
How am I supposed to know my workshop ID if I haven't published my mod yet? Not sure if you ever ran into this issue because you may have added those features after publishing your mods. 
Edit: I'm trying to utilize cim tools to add options and sliders to my mod

**Comment by PropaneDragon** [comment]
I wouldn't use the library yet, as I'm planning on putting it up on the workshop soon so you add it to your mod through there. This is because there can only be one version of CimTools in the game at one time, and if one's outdated then all of them become outdated. I've got some large updates planned for the library, so I'd wait for that or work around it for now. Sorry for that.

With regards to your problem, I think you can pass null to it and pass in the folder name for the mod name parameter to allow it to look in the mods folder.

**Comment by yenyang** [comment]
Okay I'll try to use it as an example and simplify it down to what I actually need.

**Comment by yenyang** [comment]
Thanks for the help.

**Comment by PropaneDragon** [comment]
No problem. You can copy the files out that you need if you want and use those without modification, as long as you change the namespace so it's not a part of CimTools. That might make it easier for you.

**Comment by yenyang** [comment]
Sweet thanks. This is what I'm working on if you are curious
[https://www.reddit.com/r/CitiesSkylinesModding/comments/4ehbhp/cities_skylines_rainfall/](https://www.reddit.com/r/CitiesSkylinesModding/comments/4ehbhp/cities_skylines_rainfall/)

**Comment by PropaneDragon** [comment]
No problem, and I knew I recognised your name from somewhere. Great work on the mod by the way. What bits of CimTools are you using by the way?

**Comment by yenyang** [comment]
I'm trying to add options and sliders. I want to add a slider for difficulty and one for refresh rate.

**Comment by PropaneDragon** [comment]
Ah, I see. I can't remember how up to date the one on GitHub is, but it should work fine I hope. That's one of the things I've been slowly improving.

**Comment by yenyang** [comment]
It doesn't include translations, so it's definitely not as new as the one in rush hour. I'm totally fine using a simpler version though.

**Comment by PropaneDragon** [comment]
Yeah, I've got two versions on the go because of mistakes I made early on in development for this library. It took me a while before I realised it'd cause problems if more than one mod used the same dll, so I have to slowly merge things around different versions so I don't screw anything up.

I'll have everything sorted by the time it ends up on the workshop though, and I'm glad it's providing some use to you already.
